### DISCUSSION TITLE
any way to get connection state of a paarticular node

### DISCUSSION THREAD

**Discussion by Rudrava** [xyflow/xyflow#1241] [Q&A]
I have a requirement where i need to get the connection status of a node on the node itself as i am making a app where each node needs to know its status of connection with other nodes as they would be validated upon it.
Also i would need to setup a custom flow of the nodes ie start from a particular node and read up until the end as they would perform a particular task in a linear fashion. 
![image](https://user-images.githubusercontent.com/58327905/120898547-ef51ea00-c648-11eb-829e-3f56ab5623d4.png)
this is the flow as you can see the `column selector node` needs to know about the presence of the `.csv file` node 

## any leads is appreciated thanks
Also i would update if anythings come up thanks in advance

**Comment by lkretzschmar** [comment]
So I solved this by creating a data structure (a model) that keeps a list of nodes and connectors. The react side then creates the react-flow elements (view) from this representation to render it. The model has functions like addNode() and addConnection() so it can be queried for connections that end at a certain node.

**Reply by Ark-kun** [reply]
I have solved it the same way.

>The model has functions like addNode() and addConnection() so it can be queried for connections that end at a certain node.

Stupid question (I'm a React novice), but how would you do this? Especially for function-based React components. Where would I put the methods to manipulate the state from the outside?

**Comment by Japneet15597** [comment]
> I have a requirement where i need to get the connection status of a node on the node itself as i am making a app where each node needs to know its status of connection with other nodes as they would be validated upon it.
> Also i would need to setup a custom flow of the nodes ie start from a particular node and read up until the end as they would perform a particular task in a linear fashion. 
> ![image](https://user-images.githubusercontent.com/58327905/120898547-ef51ea00-c648-11eb-829e-3f56ab5623d4.png)
> this is the flow as you can see the `column selector node` needs to know about the presence of the `.csv file` node 
> 
> ## any leads is appreciated thanks
> Also i would update if anythings come up thanks in advance

Hey, rudrava can you tell me how you got the node handles spaced evenly? Completed unrelated 😅 but any help would be great.

**Reply by Rudrava** [reply]
i basically put left and right vals on the handles check the ss for the code below
![image](https://user-images.githubusercontent.com/58327905/122330714-4c0aaa00-cf51-11eb-825a-a6d95a88674e.png)

any better way to do it would be helpful too !! 😃

**Reply by lkretzschmar** [reply]
I have an arbitrary number of handles along the vertical side of my nodes and this is how I space them: 
I reserve 10% for a header that I don't want them to use.
Note that this is not production code (yet) :-)
```
renderHandle = (type: HandleType, pos: Position, handles: []) => {
    return handles.map((port, index, arr) =>
      <Handle
        type={type}
        position={pos}
        id={port.handle}
        className={port.type}
        style={{ top: (15 + (index + 1) * (90 / (arr.length + 1))).toString() + '%' }} />
    )
  }
```

**Reply by Japneet15597** [reply]
Thanks so much both of u, my issue is that the handles on one of my node are created dynamically and i need to space them evenly regardless of how many handles are created.

**Reply by Japneet15597** [reply]
> I have an arbitrary number of handles along the vertical side of my nodes and this is how I space them:
> I reserve 10% for a header that I don't want them to use.
> Note that this is not production code (yet) :-)
> 
> ```
> renderHandle = (type: HandleType, pos: Position, handles: []) => {
>     return handles.map((port, index, arr) =>
>       <Handle
>         type={type}
>         position={pos}
>         id={port.handle}
>         className={port.type}
>         style={{ top: (15 + (index + 1) * (90 / (arr.length + 1))).toString() + '%' }} />
>     )
>   }
> ```

Similar to this

**Reply by lkretzschmar** [reply]
Mine are also created dynamically....

**Reply by Ark-kun** [reply]
Check out this MultiHandleNode PR https://github.com/wbkd/react-flow/pull/1285
